### ISSUE TITLE
[REVIEW]: datasailr - An R Package for Row by Row Data Processing, Using DataSailr Script

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3166]
**Submitting author:** @niceume (<a href="http://orcid.org/0000-0001-7015-621X">Toshihiro Umehara</a>)
**Repository:** <a href="https://github.com/niceume/datasailr" target ="_blank">https://github.com/niceume/datasailr</a>
**Version:** joss0.8.8-3
**Editor:** @fboehm
**Reviewer:** @DrMattG, @crvernon
**Archive:** <a href="https://doi.org/10.5281/zenodo.4732788" target="_blank">10.5281/zenodo.4732788</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/6ff88c7aa7cf82e7ae45daa1edf0eceb/status.svg)](https://joss.theoj.org/papers/6ff88c7aa7cf82e7ae45daa1edf0eceb)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/6ff88c7aa7cf82e7ae45daa1edf0eceb"><img src="https://joss.theoj.org/papers/6ff88c7aa7cf82e7ae45daa1edf0eceb/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/6ff88c7aa7cf82e7ae45daa1edf0eceb/status.svg)](https://joss.theoj.org/papers/6ff88c7aa7cf82e7ae45daa1edf0eceb)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/niceume/datasailr" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@DrMattG and @crvernon, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @fboehm know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/niceume/datasailr">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@niceume) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/niceume/datasailr">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@niceume) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @DrMattG and @crvernon it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.08 s (249.9 files/s, 40452.9 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
C++                              2            193            110           1385
Markdown                         2            224              0            516
R                                8             45             14            236
m4                               1             26              0            192
Bourne Shell                     5             25             19             75
C/C++ Header                     1              4              1             11
-------------------------------------------------------------------------------
SUM:                            19            517            144           2415
-------------------------------------------------------------------------------


Statistical information for the repository '5d67cab237397733aee2349c' was
gathered on 2021/04/09.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
niceume                         31          1857            153          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
niceume                    1704           91.8          1.9                6.51
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3166 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by fboehm** [comment]
@whedon generate pdf from branch joss_submission

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_submission. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: :point_left:

**Comment by fboehm** [comment]
@DrMattG and @crvernon - I think that whedon has sent you a notice inviting you to the review. I believe that you need to indicate that you accept the invitation before you can check boxes. Please let me know if you didn't get this notification. I can try to send it again.

**Comment by fboehm** [comment]
@whedon list commands

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by fboehm** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by DrMattG** [comment]
> @DrMattG and @crvernon - I think that whedon has sent you a notice inviting you to the review. I believe that you need to indicate that you accept the invitation before you can check boxes. Please let me know if you didn't get this notification. I can try to send it again.

@fboehm can you send it again - I thought I had accepted it but I can not check the boxes.

**Comment by crvernon** [comment]
@fboehm @DrMattG I also get:
```
Sorry, we couldn't find that repository invitation. It is possible that the invitation was revoked or that you are not logged into the invited account.
```

when visiting https://github.com/openjournals/joss-reviews/invitations 

I am signed in.  Thanks.

**Comment by fboehm** [comment]
Thanks for the prompt replies, @DrMattG and @crvernon ! I think it was my error when initially inviting you via whedon. I must have used the wrong syntax. Let me "reinvite" you.

**Comment by fboehm** [comment]
@whedon re-invite @DrMattG as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@drmattg please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by fboehm** [comment]
@whedon re-invite @crvernon as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@crvernon please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by niceume** [comment]
@whedon generate pdf from branch joss_submission

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_submission. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: :point_left:

**Comment by niceume** [comment]
@whedon generate pdf from branch joss_submission

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_submission. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: :point_left:

**Comment by crvernon** [comment]
@fboehm I believe that @niceume has addressed all of the issues I submitted in the `joss_submission`branch of the `datasailr` repository here:  https://github.com/niceume/datasailr/tree/joss_submission 

I have checked all of my review boxes and consider this submission as ready awaiting the approval of you and @DrMattG 

Excellent work @niceume !

**Comment by niceume** [comment]
Thank you very much for your review, @crvernon !

**Comment by DrMattG** [comment]
> @fboehm I believe that @niceume has addressed all of the issues I submitted in the `joss_submission`branch of the `datasailr` repository here: https://github.com/niceume/datasailr/tree/joss_submission
> 
> I have checked all of my review boxes and consider this submission as ready awaiting the approval of you and @DrMattG
> 
> Excellent work @niceume !

I agree - great work @niceume. I am happy that the issues I raised are now addressed. Thanks @fboehm for the opportunity to review - it's a really nice process.

**Comment by niceume** [comment]
Thank you very much, @DrMattG ! The paper and the software have become improved much.

**Comment by niceume** [comment]
@whedon generate pdf from branch joss_submission

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_submission. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: :point_left:

**Comment by niceume** [comment]
@fboehm

* I have checked the generated wheedon PDF proof
* I created a release on GitHub for this version ( https://github.com/niceume/datasailr/releases/tag/joss0.8.8 )
* I have created a Zenodo-archive of this release.

The Zenodo doi is 10.5281/zenodo.4716943

Thank you.

**Comment by fboehm** [comment]
@niceume - before we proceed, a few minor fixes are needed for the paper. 

line 15 and other lines: row-wise (or rowwise - I believe that both spellings are acceptable)
line 20: domain-specific
throughout: insert spaces before parentheses around citations. So it should look like "we used R (R foundation, 2021)", where there's a space between R and (. 
line 37: Do you want "datasailr" as the first word of the sentence? 
line 70: Do you want "dplyr" instead of "Dplyr"? You can add curly braces around dplyr, like "{dplyr}", in the bib file to force rendering of lower-case letters.
line 66: same comment - do you want "sqldf"? and "R" is needed in place of "r".

Please make these fixes and double-check the resulting pdf.

**Comment by niceume** [comment]
@whedon generate pdf from branch joss_submission

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_submission. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: :point_left:

**Comment by niceume** [comment]
@whedon generate pdf from branch joss_submission

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_submission. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: :point_left:

**Comment by niceume** [comment]
@fboehm - Thank you very much for your editing.  I have updated the article.

**Comment by fboehm** [comment]
Thank you, @niceume, for making the edits. 

Can you please update the hyperlink to the github release? When I click on the current link, I get a 404 error.

**Comment by fboehm** [comment]
Also, please share a hyperlink to the archive on zenodo.

**Comment by fboehm** [comment]
I re-read the paper pdf, and I'm quite pleased with the changes that you made. I think it's ready for publication.

**Comment by fboehm** [comment]
@niceume - I'll set the archive (with zenodo doi) and the version number once you update the links.

**Comment by niceume** [comment]
Thank you very much, @fboehm !
I deleted the previous tag and release. I have made a new tag and release for the last update commit.
The Zenodo DOI for the final version is 10.5281/zenodo.4718464

**Comment by fboehm** [comment]
@whedon set joss0.8.8-2 as version

**Comment by whedon** [comment]
OK. joss0.8.8-2 is the version.

**Comment by fboehm** [comment]
@whedon set 10.5281/zenodo.4718464 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4718464" target="_blank">10.5281/zenodo.4718464</a> is the archive.

**Comment by fboehm** [comment]
Never mind - I found the version number in the archive.

**Comment by fboehm** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3166 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by fboehm** [comment]
@whedon accept from branch joss_submission

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2261

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2261, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss_submission 
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by niceume** [comment]
@fboehm - Thank you for proceeding! 
The DOI and the tag are correct. 
I have checked the final proof. Everything looks fine!

**Comment by Kevin-Mattheus-Moerman** [comment]
@niceume I have proodread you paper and have some minor comments to address before we can move on. Often the word `a` or `the` seemed missing from a sentence. Please implement these changes or respond if you disagree. Thanks. 

- [ ] Can you rephrase the sentence `I present yet another way to process data in a row by row manner using data manipulation oriented script, DataSailr script.`, first of all please remove all occurrences of "I" (or "we" etc if any) and rephrase. Furthermore I think the word "a" should be added, for instance I propose: `This work presents another way to process data in a row by row manner using a data manipulation oriented script, DataSailr script.`
- [ ] Fix `Row-wise approach does not require vectors,....`, which should likely read `The row-wise approach does not require vectors,....`
- [ ] Fix `DataSailr script is a domain-specific language...`, which should likely read `The DataSailr script is a domain-specific language...`
- [ ] Fix `...the result vector to BMI column`, which should likely read `...the result vector to a BMI column`.
- [ ] Fix `... a main function called sail() function`, which should likely read `... a main function called sail()`.
- [ ] Fix `The first argument is data frame.`, which should likely read `The first argument is a data frame.`
- [ ] Fix `The second argument is DataSalr script,`, which should likely read `The second argument is the DataSalr script,`
- [ ] Fix `explained in README and DataSailr website`, which should likely read `explained in the README and on the DataSailr website`

Suggestion
- [ ] I notice your hearth felt and kind personal acknowledgement. This is the first time I've seen such a personal acknowledgement. Although I can identify with the sentiment, I do kindly ask if you could consider to only acknowledge professional/academic support. This can include financial support, technical support, advise, or the acknowledgement of contributors who are not authors on the paper.

**Comment by niceume** [comment]
@whedon generate pdf from branch joss_submission

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_submission. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03166/joss.03166/10.21105.joss.03166.pdf) :page_facing_up: :point_left:

**Comment by niceume** [comment]
@Kevin-Mattheus-Moerman - Thank you very much for your reading and comments. I have fixed the parts you mention. I also understand that this is an academic/professional work, and the personal acknowledgement part is now removed from the paper.

@fboehm - If this looks ok, I will make a new release, archive it to Zonodo and obtain a new DOI.

**Comment by Kevin-Mattheus-Moerman** [comment]
@niceume thanks for implementing those changes. I checked them and I think the paper looks good now. Since these are only editorial changes, from our perspective you do not need to create an updated release and archived version. Therefore if you wish I can process acceptance now.

If you do want the paper in the archived version to be up to date then feel free to do as you proposed. Let us know if the version tag changes too. I'll then update the archive link and the version tag and process acceptance at that point.

**Comment by niceume** [comment]
@Kevin-Mattheus-Moerman - Thank you very much for having things proceed.

The release tag and DOI are updated as follows.
The release tag is joss0.8.8-3
The Zonodo DOI is 10.5281/zenodo.4732788

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set version as joss0.8.8-3

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set 10.5281/zenodo as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo" target="_blank">10.5281/zenodo</a> is the archive.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set joss0.8.8-3 as version

**Comment by whedon** [comment]
OK. joss0.8.8-3 is the version.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set 10.5281/zenodo.4732788 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4732788" target="_blank">10.5281/zenodo.4732788</a> is the archive.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3166 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept from branch joss_submission

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2272

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2272, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss_submission 
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept deposit=true from branch joss_submission

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1388777955514621955) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2273
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03166](https://doi.org/10.21105/joss.03166)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by Kevin-Mattheus-Moerman** [comment]
Congratulations @niceume your paper has now been published in JOSS! 

Thank you @fboehm for editing this work, and thank you very much @DrMattG and @crvernon for your review efforts!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03166/status.svg)](https://doi.org/10.21105/joss.03166)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03166">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03166/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03166/status.svg
   :target: https://doi.org/10.21105/joss.03166
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03166/status.svg)](https://doi.org/10.21105/joss.03166)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by niceume** [comment]
@fboehm @DrMattG  @crvernon @Kevin-Mattheus-Moerman  
I really appreciate your helps. I am really excited! Thank you very much!!
